tensorflow/core/kernels/dml_pooling_ops.cc

```diff
@@ -568,13 +568,22 @@ using DmlMaxPoolKernel =
       Name("MaxPoolGrad").Device(DEVICE_DML).TypeConstraint<type>("T"), \
       DmlKernelWrapper<DmlMaxPoolGradKernel,                            \
                        GetOutputShapeAsInputShapeHelper>);              \
-  REGISTER_KERNEL_BUILDER(Name("MaxPoolGradV2")                         \
-                              .Device(DEVICE_DML)                       \
-                              .TypeConstraint<type>("T")                \
-                              .HostMemory("ksize")                      \
-                              .HostMemory("strides"),                   \
-                          DmlKernelWrapper<DmlMaxPoolGradKernel,        \
-                                           GetOutputShapeAsInputShapeHelper>);
+  REGISTER_KERNEL_BUILDER(                                              \
+      Name("MaxPoolGradV2")                                             \
+          .Device(DEVICE_DML)                                           \
+          .TypeConstraint<type>("T")                                    \
+          .HostMemory("ksize")                                          \
+          .HostMemory("strides"),                                       \
+      DmlKernelWrapper<DmlMaxPoolGradKernel,                            \
+                       GetOutputShapeAsInputShapeHelper>);              \
+  REGISTER_KERNEL_BUILDER(                                              \
+      Name("MaxPool3DGrad")                                             \
+          .Device(DEVICE_DML)                                           \
+          .TypeConstraint<type>("T")                                    \
+          .TypeConstraint<type>("TInput"),                              \
+      DmlKernelWrapper<DmlMaxPoolGradKernel,                            \
+                       GetOutputShapeAsInputShapeHelper>);              \
+
```

Comment:
nit: I think this needs to be clang formatted (all lines were already less than 80 characters before)